app/controllers/ConvertedPackagingCreditController.scala

```diff
@@ -44,34 +49,60 @@ class ConvertedPackagingCreditController @Inject() (
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController with I18nSupport {
 
-  val form = formProvider()
+  private val form = formProvider()
 
-  def onPageLoad(mode: Mode): Action[AnyContent] =
-    (identify andThen getData andThen requireData) {
+  def onPageLoad(mode: Mode): Action[AnyContent] = {
+    (identify andThen getData andThen requireData).async {
       implicit request =>
         val preparedForm = request.userAnswers.get(ConvertedPackagingCreditPage) match {
-          case None        => form
+          case None => form
           case Some(value) => form.fill(value)
         }
 
-        Ok(view(preparedForm, mode))
+        val futureCreditBalanceAvailable: Future[Option[String]] = exportCreditBalanceAvailable(request)
+        futureCreditBalanceAvailable.map {
+          possibleCreditBalanceAvailable => Ok(view(preparedForm, mode, possibleCreditBalanceAvailable))
+        }
+
     }
+  }
+
+  private def exportCreditBalanceAvailable(request: DataRequest[AnyContent])(implicit hc: HeaderCarrier): Future[Option[String]] = {
+    val obligation = request.userAnswers.get[TaxReturnObligation](ObligationCacheable).getOrElse(
+      throw new IllegalStateException("Obligation not found in user-answers")
+    )
+
+    exportCreditsConnector.get(request.request.pptReference,
+      obligation.fromDate.minusYears(2),
+      obligation.fromDate.minusDays(1)
+    )
+  }.map {
+    case Right(balance) => Some(displayMonetaryValue(balance.totalExportCreditAvailable))
+    case Left(_) => None
+  }
+
+  def displayMonetaryValue(money: BigDecimal): String = s"£${money.setScale(2, RoundingMode.HALF_EVEN)}"
```

Comment:
this is already a method on bigdecial in the viewmodel package i believe `moneyt.asPounds`

Reply:
Fair enough. This line is just copied from 'other' branch

Reply:
sorted